language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/definition/DefinitionUtil.java

```diff
@@ -63,25 +75,80 @@
     }
 
     private static Optional<Location> getLocation(Symbol symbol, DocumentServiceContext context) {
-        if (symbol.getLocation().isEmpty()) {
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty() || symbol.getModule().isEmpty()) {
             return Optional.empty();
         }
+        ModuleID moduleID = symbol.getModule().get().id();
+        String orgName = moduleID.orgName();
+        String moduleName = moduleID.moduleName();
 
+        Optional<Path> filepath;
+        if (CommonUtil.isLangLib(orgName, moduleName)) {
+            filepath = getFilePathForLanglib(orgName, moduleName, project.get(), symbol);
+        } else {
+            filepath = getFilePathForDependency(orgName, moduleName, project.get(), symbol);
+        }
+
+        if (filepath.isEmpty()) {
+            return Optional.empty();
+        }
+
+        String uri = filepath.get().toUri().toString();
         io.ballerina.tools.diagnostics.Location symbolLocation = symbol.getLocation().get();
         LinePosition startLine = symbolLocation.lineRange().startLine();
         LinePosition endLine = symbolLocation.lineRange().endLine();
         Position start = new Position(startLine.line(), startLine.offset());
         Position end = new Position(endLine.line(), endLine.offset());
         Range range = new Range(start, end);
 
-        if (symbol.getModule().isEmpty()) {
-            return Optional.empty();
+        return Optional.of(new Location(uri, range));
+    }
+
+    private static Optional<Path> getFilePathForDependency(String orgName, String moduleName,
+                                                           Project project, Symbol symbol) {
+        Collection<ResolvedPackageDependency> dependencies =
+                project.currentPackage().getResolution().dependencyGraph().getNodes();
+        Path filepath = null;
+        String sourceFile = symbol.getLocation().get().lineRange().filePath();
+        for (ResolvedPackageDependency depNode : dependencies) {
+            Package depPackage = depNode.packageInstance();
+            for (ModuleId moduleId : depPackage.moduleIds()) {
+                if (depPackage.packageOrg().value().equals(orgName) &&
+                        depPackage.module(moduleId).moduleName().toString().equals(moduleName)) {
+                    Module module = depPackage.module(moduleId);
+                    for (DocumentId docId : module.documentIds()) {
+                        if (module.document(docId).name().equals(sourceFile)) {
+                            filepath =
+                                    module.project().documentPath(docId).orElseThrow();
```

Comment:
We could do `return module.project().documentPath(docId).flatMap(Optional::of);` instead of throwing.

Reply:
Used an alternative approach instead of using the flatmap